src/content/docs/en/guides/environment-variables.mdx

```diff
@@ -9,38 +9,240 @@ Astro supports 2 main ways for handling environment variables: the `astro:env` m
 
 ## `astro:env`
 
-- feature tldr
-- example usage
+`astro:env` lets you configure a type-safe schema for your environment variables, and indicate whether they should be available on the server or the client.
+
+To configure a schema, add the `env` option to your Astro config and define your client and server variables:
+
+```js title="astro.config.mjs" ins={4-10}
+import { defineConfig, envField } from 'astro/config'
+
+export default defineConfig({
+  env: {
+    schema: {
+      API_URL: envField.string({ context: "client", access: "public", optional: true }),
+      PORT: envField.number({ context: "server", access: "public", default: 4321 }),
+      API_SECRET: envField.string({ context: "server", access: "secret" }),
+    }
+  }
+})
+```
+
+You can import and use your defined variables from the appropriate `/client` or `/server` module:
+
+```astro
+---
+import { API_URL } from "astro:env/client"
+import { API_SECRET_TOKEN } from "astro:env/server"
+const data = await fetch(`${API_URL}/users`, {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    "Authorization": `Bearer ${API_SECRET_TOKEN}`
+  },
+});
+---
+
+<script>
+import { API_URL } from "astro:env/client"
+fetch(`${API_URL}/ping`)
+</script>
+```
 
-TODO: where to talk about integrations & adapters?
+**TODO: where to talk about integrations & adapters?**
```

Comment:
If you mean for integration/adapter *authors*, then this can be a note with a link to one of the other API pages. I don't think this page should have information about writing integrations.